### PR TITLE
ALFLayout Generator

### PR DESCRIPTION
**Pull request by knute-sifive**
New python script that replaces the old, outdated perl generator.

This new generator consumes a `location.dat` file from a generated pipeout and, when coupled with a custom python script, can be used to explicitly setup ALF files used by the Argos tools.

See `pydoc ALFLayout` for more information as well as the example in example.CoreModel

### CODE REVIEW

File: sparta/example/CoreModel/gen_layouts.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+import sys,os
+
+sys.path.append('../../../../helios/pipeViewer/scripts')
```
Comment (kathlenehurt-sifive) [inline sparta/example/CoreModel/gen_layouts.py:3]: Is it possible to specify an absolute path here with a variable instead of the relative path?

**Comment by kathlenehurt-sifive** [comment]
I'd recommend adding a main function in `gen_layouts.py` incase you ever want to add command line parameters.

File: sparta/example/CoreModel/gen_layouts.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        ./sparta_core_example -i1 -z pipeout/
+    ''')
+    exit(255)
+
+try:
+    os.mkdir('layouts')
+except:
+    pass
+
+num_cycles = 55
```
Comment (kathlenehurt-sifive) [inline sparta/example/CoreModel/gen_layouts.py:19]: Variables intended to be constant should be named in all caps.

**Review by kathlenehurt-sifive — APPROVED** [review]
Neat!

File: sparta/example/CoreModel/gen_layouts.py (hunk 1 above)
Comment (knute-sifive) [reply sparta/example/CoreModel/gen_layouts.py:3]: Yeah, I'll change it to build the abs path.

File: sparta/example/CoreModel/gen_layouts.py (hunk 2 above)
Comment (knute-sifive) [reply sparta/example/CoreModel/gen_layouts.py:19]: Done

File: sparta/example/CoreModel/gen_layouts.py (hunk 1 above)
Comment (knute-sifive) [reply sparta/example/CoreModel/gen_layouts.py]: Doine

**Comment by knute-sifive** [comment]
> I'd recommend adding a main function in `gen_layouts.py` incase you ever want to add command line parameters.

Not a bad idea.  Wish I were a better python programmer.  Not sure what I'd do with a `main`.  😄

**Pull request merged**